compiler/parser/chapel.ypp

```diff
@@ -2222,6 +2226,14 @@ literal_expr:
     {
       $$ = new CallExpr("chpl__buildAssociativeArrayExpr", $2);
     }
+| str_bytes_literal TLSBR opt_actual_ls TRSBR
```

Comment:
I have some doubts whether we should use `opt_actual_ls` here or something more specific. For one, it'll allow `"foo"[]` to be parsed as well as `"foo"[1,2]`. 

OTOH, I am not sure if there is anything else readily available that we can use in the parser. Arguably this is a bit of a special case.

@bradcray -- do you have an opinion here?

Reply:
I agree that it'd be better to make this a tighter filter if possible.  Walking the grammar from opt_actual_ls, I'm wondering whether we could use `actual_expr` and have its action be similar to the one in `actual_ls`: `new CallExpr(PRIM_ACTUALS_LIST, $1);`.

Reply:
That may be better, right. That would allow named actuals, but I don't see that as a huge deal. Technically you should be able to use named actuals while calling `this` via indexing, though it feels a bit wrong.

Reply:
Oh, that's a good point (and arguably, we already have this issue for other array-style accesses, right?).  We've discussed having a way of saying "this argument can't be named" in the language, and if/when we add such a feature, I think these access routines would be an obvious place to use them.  Of course, that would still permit them to be parsed, but that may be OK since it would generate a better error (presumably) than simply "syntax error".  Alternatively, we could refactor the grammar to generate these better errors here and now, though I think that's independent of this effort since we've already got that issue (?).

Reply:
For an array, I got an unresolved access error. For a string it compiled and ran fine. I am not sure whether I like either behavior and probably prefer this to be part of the syntax and give a syntax error.

This is definitely not a part of this effort, though. I think `actual_expr` is as good as we can get without any other effort.